samples/cpp/image_generation/stable_diffusion_export_import.cpp

```diff
@@ -38,7 +38,7 @@ void dedicated_models_export_import(const std::filesystem::path& root_dir) {
     auto unet = ov::genai::UNet2DConditionModel(root_dir / "unet", device);
     unet.export_model(blob_path / "unet");
 
-    auto vae = ov::genai::AutoencoderKL(root_dir / "vae_decoder", "CPU", ov::AnyMap{});
+    auto vae = ov::genai::AutoencoderKL(root_dir / "vae_decoder", device, ov::AnyMap{});
     vae.export_model(blob_path);
```

Comment:
Python still needs a fix

Reply:
`AutoencoderKL` has device in python https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/python/image_generation/stable_diffusion_export_import.py#L47

Reply:
Why is `compile()` explicit there?

Reply:
In both, python and c++. I'll take a look, but not sure when